homeassistant/components/roborock/vacuum.py

```diff
@@ -96,6 +97,22 @@ async def async_setup_entry(
         supports_response=SupportsResponse.NONE,
     )
 
+    platform.async_register_entity_service(
+        VACUUM_CLEAN_ROOMS_SERVICE_NAME,
+        cv.make_entity_service_schema(
+            {
+                vol.Required("room_ids"): vol.Any(
+                    vol.Coerce(int), [vol.Coerce(int)], vol.Coerce(str)
+                ),
```

Comment:
I am a bit unsure if it works but could we maybe use the `SelectSelector` here and construct it with the proper values and room names?